packages/squiggle-lang/src/analysis/Node.ts

```diff
@@ -0,0 +1,68 @@
+import { LocationRange } from "../ast/types.js";
+import { Type } from "../types/Type.js";
+import {
+  AnyTypedExpressionNode,
+  AnyTypedStatementNode,
+  expressionKinds,
+  statementKinds,
+  TypedASTNode,
+} from "./types.js";
+
+export abstract class Node<T extends string> {
```

Comment:
We have the separate class hierarchy for typed AST, while the old AST is still represented as basic JS objects.

I'm not sure that this is optimal; it causes more code duplication than I'd like.
 
But fixing this is another 1-2 days task, and there are some details why it's not entirely straightforward, so I'll do it some time later.